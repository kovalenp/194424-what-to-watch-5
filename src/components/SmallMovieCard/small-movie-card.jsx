import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

function SmallMovieCard({
  id,
  title,
  img,
  onMouseEnterHandler,
  onMouseLeaveHandler,
}) {
  return (
    <article className="small-movie-card catalog__movies-card">
      <div
        className="small-movie-card__image"
        onMouseEnter={onMouseEnterHandler}
        onMouseLeave={onMouseLeaveHandler}
        id={id}
      >
        <img src={img.tile} alt={title} width="280" height="175" />
      </div>
      <h3 className="small-movie-card__title">
        <Link className="small-movie-card__link" to={`films/${id}`}>
          {title}
        </Link>
      </h3>
    </article>
  );
}

SmallMovieCard.propTypes = {
  title: PropTypes.string.isRequired,
  img: PropTypes.shape({
    poster: PropTypes.string,
    tile: PropTypes.string.isRequired,
  }),
  onMouseEnterHandler: PropTypes.func.isRequired,
  onMouseLeaveHandler: PropTypes.func.isRequired,
  id: PropTypes.number,
};

export default SmallMovieCard;
