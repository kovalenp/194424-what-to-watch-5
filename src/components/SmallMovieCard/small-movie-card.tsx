import React from "react";
import PropTypes from "prop-types";

function SmallMovieCard({ title, img, onMouseOverHandler }) {
  return (
    <article className="small-movie-card catalog__movies-card">
      <div className="small-movie-card__image" onMouseOver={onMouseOverHandler}>
        <img src={img.tile} alt={title} width="280" height="175" />
      </div>
      <h3 className="small-movie-card__title">
        <a className="small-movie-card__link" href="movie-page.html">
          {title}
        </a>
      </h3>
    </article>
  );
}

SmallMovieCard.propTypes =
  PropTypes.shape({
    title: PropTypes.string.isRequired,
    img: PropTypes.shape({
      poster: PropTypes.string,
      tile: PropTypes.string.isRequired,
    }),
    onMouseOverHandler: PropTypes.func.isRequired,
  };


export default SmallMovieCard;
