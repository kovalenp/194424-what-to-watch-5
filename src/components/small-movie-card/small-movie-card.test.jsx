import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter} from 'react-router-dom';

import SmallMovieCard from "./small-movie-card";
import movie from "../../mocks/movie";

describe(`SmallMovieCard meets B21`, () => {

  it(`and rendered correctly for auth'ed`, () => {
    const tree = renderer.create((
      <MemoryRouter>
        <SmallMovieCard
          movie={movie}
          active={``}
          onActiveChange={() => { }} />
      </MemoryRouter>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
