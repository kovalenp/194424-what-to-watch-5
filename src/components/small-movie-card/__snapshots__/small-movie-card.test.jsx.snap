// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallMovieCard meets B21 and rendered correctly for auth'ed 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <div
    className="small-movie-card__image"
    id={1}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <video
      controls={false}
      height="175"
      muted={true}
      poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
      preload="none"
      src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      Bronson
    </a>
  </h3>
</article>
`;
